issue_drafts/issue_2.md

```diff
@@ -0,0 +1,63 @@
+# Issue 2: Reading Snakefile Parameters into Your Script
+
+Pipeline steps: No new pipeline steps
+Concepts learned: Snakefile python module to read in variables from snakefile
+
+## Defining Named Variables in a Snakefile
+In the last issue, we wrote a Snakefile rule to fetch a zipped folder containing temperature prediction data:
+```
+rule get_sb_data:
+    output:
+        1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip
+    script:
+        1_fetch/sb_get.py
+```
+
+There were no file inputs to this step (if there were, we would have defined them as a Snakefile 'input'). However, if you look at the `1_fetch/sb_get.py` script, you will notice that there are two pieces of information that could be considered 'inputs' to this pipeline step - the ScienceBase item that we are fetching data from, and the name of the file we want to download. These 'inputs' have been hardcoded into the script as variables:
+```
+if __name__ == "__main__":
+    sb_item = "5e5d0bb9e4b01d50924f2b36"
+    sb_file = "1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip"
+    main(sb_item, sb_file)
+"
+```
+
+It can be helpful to specify input variables for your pipeline step in the Snakefile. Specifying these varaiables in the Snakefile allows you to get a more comprehensive view of the inputs and outputs for your pipeline step. It also allows you to more easily update your pipeline if, for example, you were to switch to a different (but identically structured) set input data from ScienceBase.
+
+Let's define these variables in the Snakefile now. The snakemake Python module (which we will use to read our variables into the script) has access to the information stored as inputs, outputs, or params in the Snakefile. We have already defined the string stored in the `sb_file` variable in the output of our current rule. Let's give this string a name so that we can reference it by name in our script:
+```
+rule get_sb_data:
+    output:
+        sb_file = "1_fetch/out/pgdl_predictions_04_N45.50-48.00_W92.00-93.00.zip"
+    script:
+        "1_fetch/sb_get.py"
+```
+
+We have not defined the string stored in the `sb_item` variable of our Python script anywhere in our Snakefile yet. Since it is not an input or output file for the script called in this step, we will define this variable as a parameter for this rule:
```

Comment:
```suggestion
We have not defined the string stored in the `sb_item` variable of our Python script anywhere in our Snakefile yet. Since it is not an input or output file for the script called in this step, we will define this variable as a parameter for this rule by using the `params` directive:
```

Reply:
The `params` directive is introduced for the first time here, so I though the text could draw more attention to it. We could also consider another sentence describing the directive: something like "The `params` directive is useful for providing any information other than input and output files to the script." It's redundant with what you've already said, though, so maybe it's not worth including.